ocs_ci/deployment/vmware.py

```diff
@@ -24,23 +32,302 @@ def __init__(self):
         """
         super(VSPHEREBASE, self).__init__()
         self.region = config.ENV_DATA.get('region')
+        self.server = config.ENV_DATA.get('vsphere_server')
+        self.user = config.ENV_DATA.get('vsphere_user')
+        self.password = config.ENV_DATA.get('vsphere_password')
+        self.cluster = config.ENV_DATA.get('vsphere_cluster')
+        self.datacenter = config.ENV_DATA.get('vsphere_datacenter')
+        self.datastore = config.ENV_DATA.get('vsphere_datastore')
+        self.vsphere = VSPHEREUtil(self.server, self.user, self.password)
+
+    def attach_disk(self, size=100):
+        """
+        Add a new disk to all the workers nodes
+
+        Args:
+            size (int): Size of disk in GB (default: 100)
+
+        """
+        vms = self.vsphere.get_all_vms_in_pool(
+            config.ENV_DATA.get("cluster_name"),
+            self.datacenter,
+            self.cluster
+        )
+        # Add disks to all worker nodes
+        for vm in vms:
+            if "compute" in vm.name:
+                self.vsphere.add_disks(
+                    config.ENV_DATA.get("extra_disks", 1),
+                    vm,
+                    size,
+                    constants.VM_DISK_TYPE
+                )
 
 
 class VSPHEREUPI(VSPHEREBASE):
     """
     A class to handle vSphere UPI specific deployment
     """
     def __init__(self):
-        self.name = self.__class__.__name__
         super(VSPHEREUPI, self).__init__()
+        self.ipam = config.ENV_DATA.get('ipam')
+        self.token = config.ENV_DATA.get('ipam_token')
+        self.cidr = config.ENV_DATA.get('machine_cidr')
+        self.vm_network = config.ENV_DATA.get('vm_network')
 
     class OCPDeployment(BaseOCPDeployment):
         def __init__(self):
             super(VSPHEREUPI.OCPDeployment, self).__init__()
+            self.public_key = {}
+            self.upi_repo_path = os.path.join(
+                constants.EXTERNAL_DIR,
+                'installer'
+            )
+            self.terraform_work_dir = f"{self.upi_repo_path}/upi/vsphere/"
+            self.terraform = Terraform(self.terraform_work_dir)
+
+        def deploy_prereq(self):
+            """
+            Pre-Requisites for vSphere UPI Deployment
+            """
+            super(VSPHEREUPI.OCPDeployment, self).deploy_prereq()
+            # create ignitions
+            self.create_ignitions()
+            self.kubeconfig = f"{self.cluster_path}/auth/kubeconfig"
+
+            # git clone repo from openshift installer
+            clone_repo(
+                constants.VSPHERE_INSTALLER_REPO, self.upi_repo_path
+            )
+
+            # upload bootstrap ignition to public access server
+            bootstrap_path = f"{config.ENV_DATA.get('cluster_path')}/{constants.BOOTSTRAP_IGN}"
```

Comment:
As mentioned above, I think `os.path.join(...)` is preferred for path concatenation.